src/analysis/lattice.jl

```diff
@@ -148,33 +192,52 @@ function Base.show(io::IO, inc::Incidence)
         else
             print(io, minus ? " - " : " + ")
         end
+    time = inc.row[1]
+    time_linear = time !== nonlinear
+    is_grouped(v, i) = isa(v, Linearity) && (v.state_dependent || (v.time_dependent || i == 1) && in(time, (linear_state_dependent, nonlinear)))
     for (i, v) in zip(rowvals(inc.row), nonzeros(inc.row))
-        v !== nonlinear || continue
-        print_plusminus(io, v < 0)
-        if abs(v) != 1 || i == 1
-            print(io, abs(v))
+        is_grouped(v, i) && continue
+        if isa(v, Float64)
+            print_plusminus(io, v < 0)
+            if abs(v) != 1
+                print(io, abs(v))
+            end
+        else
+            !first && print(io, " + ")
+            first = false
+            if !is_grouped(inc.row[1], 1)
+                ᵢ = i > 1 ? subscript(i - 1) : 'ₜ'
+                if v.time_dependent
+                    print(io, time_linear ? "∝t" : "f$ᵢ(t)", " * ")
```

Comment:
It may be clearer to show this a bit more explicitly as `f(u)*t + f(u,t)*u₁` etc. instead of introducing `∝`

Reply:
It would be less precise as `u := ∑ᵢuᵢ`, while in e.g. `u₁ + t*u₂*u₃ + u₄` we know the structure has to be `u₁ + f(∝t, ∝u₂, ∝u₃) + u₄`: we can see which states don't have themselves any state dependence and remove them from `u`. If we printed that as `f(u)*t + u₁ + f(u, t)*u₂ + f(u, t)*u₃ + u₄`, I don't think we would gain clarity and this would be a bit misleading as we might think `u₁` and `u₄` could be in `u` which might make them nonlinear in other states or time.

Reply:
Oh! I understand this now - I didn't realize this was being used as "at-most proportionally dependent" (I thought it was a new constant and expressing non-linearity rather than linearity)

I agree, that's a nice shorthand actually 👍

Is it possible to make it fully-precise w.r.t. time-dependence?
```julia
julia> inc = (@incidence u₁ *ᵢ u₃+ u₂ *ᵢ t).row
4611686018427387903-element SparseVector{Union{Float64, Linearity}, Int64} with 4 stored entries:
  [1]  =  Linearity(false, true, false)
  [2]  =  Linearity(false, true, false)
  [3]  =  Linearity(true, false, false)
  [4]  =  Linearity(false, true, false)

julia> inc = (@incidence u₁ *ᵢ u₃+ u₂ *ᵢ t)
Incidence(f(∝t, ∝u₁, ∝u₂, ∝u₃))
```

Reply:
Implemented in https://github.com/JuliaComputing/DAECompiler.jl/pull/41/commits/b1d0e496388a3dcf4347adc2551eb260932c3773, seems to be fine.